app/src/main/java/com/example/mywatchface/mvp/view/WatchFaceView.kt

```diff
@@ -0,0 +1,222 @@
+    }
+
+    override fun initGrayBackgroundBitmap() {
+        mGrayBackgroundBitmap = Bitmap.createBitmap(
+            mBackgroundBitmap.width,
+            mBackgroundBitmap.height,
+            Bitmap.Config.ARGB_8888
+        )
+        val canvas = Canvas(mGrayBackgroundBitmap)
+        val grayPaint = Paint()
+        val colorMatrix = ColorMatrix()
+        colorMatrix.setSaturation(ZERO_FLOAT)
+        val filter = ColorMatrixColorFilter(colorMatrix)
+        grayPaint.colorFilter = filter
+        canvas.drawBitmap(mBackgroundBitmap, ZERO_FLOAT, ZERO_FLOAT, grayPaint)
+    }
+
+    override fun getRandomColor(): Int {
+        val random = Random()
+        return Color.argb(
+            ALPHA, random.nextInt(RGB_POSSIBLE_VALUES),
+            random.nextInt(RGB_POSSIBLE_VALUES),
+            random.nextInt(RGB_POSSIBLE_VALUES)
+        )
+    }
+
+    override fun drawWatchFace(canvas: Canvas, mAmbient: Boolean) {
+        val innerTickRadius = mCenterX - INNER_TICK_RADIUS_OFFSET
+        val outerTickRadius = mCenterX
+        for (tickIndex in HOURS) {
+            val tickRot = (tickIndex.toDouble() * ONE_ROUND / HOURS_IN_ANALOG_CLOCK).toFloat()
+            val innerX = sin(tickRot.toDouble()).toFloat() * innerTickRadius
+            val innerY = (-cos(tickRot.toDouble())).toFloat() * innerTickRadius
+            val outerX = sin(tickRot.toDouble()).toFloat() * outerTickRadius
+            val outerY = (-cos(tickRot.toDouble())).toFloat() * outerTickRadius
+            canvas.drawLine(
+                mCenterX + innerX, mCenterY + innerY,
+                mCenterX + outerX, mCenterY + outerY, mTickAndCirclePaint
+            )
+        }
+        val seconds = mCalendar.get(Calendar.SECOND) + mCalendar.get(Calendar.MILLISECOND) / MILLISECONDS
+        val secondsRotation = seconds * FRACTION_ANGLE
+        val minutesRotation = mCalendar.get(Calendar.MINUTE) * FRACTION_ANGLE
+        val hourHandOffset = mCalendar.get(Calendar.MINUTE) / HOUR_HAND_OFFSET
+        val hoursRotation = mCalendar.get(Calendar.HOUR) * OFFSET + hourHandOffset
+        canvas.save()
+        canvas.rotate(hoursRotation, mCenterX, mCenterY)
+        canvas.drawLine(
+            mCenterX,
+            mCenterY - CENTER_GAP_AND_CIRCLE_RADIUS,
+            mCenterX,
+            mCenterY - sHourHandLength,
+            mHourPaint
+        )
+        canvas.rotate(minutesRotation - hoursRotation, mCenterX, mCenterY)
+        canvas.drawLine(
+            mCenterX,
+            mCenterY - CENTER_GAP_AND_CIRCLE_RADIUS,
+            mCenterX,
+            mCenterY - sMinuteHandLength,
+            mMinutePaint
+        )
+        if (!mAmbient) {
+            canvas.rotate(secondsRotation - minutesRotation, mCenterX, mCenterY)
+            canvas.drawLine(
+                mCenterX,
+                mCenterY - CENTER_GAP_AND_CIRCLE_RADIUS,
+                mCenterX,
+                mCenterY - mSecondHandLength,
+                mSecondPaint
+            )
+
```

Comment:
pls remove empty spaces

Reply:
done

Reply:
Pls the empty space is still there

Reply:
i'm sorry, i didnt see that space, nwo its removed too